### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by victor-paumier**

### CODE REVIEW

File: .docker/config/vhosts/buddysystem.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
-Define server_name buddysystem.local
-Define basedocroot  /home/wwwroot/buddysystem
+Define server_name  buddysystem.local
+Define basedocroot  /var/www/buddysystem
```
Comment (JeremieSamson) [inline .docker/config/vhosts/buddysystem.conf]:
```suggestion
Define basedocroot /var/www/buddysystem
```

File: .docker/config/vhosts/buddysystem.conf [hunk 2]
```diff
@@ -1,5 +1,5 @@
-Define server_name buddysystem.local
-Define basedocroot  /home/wwwroot/buddysystem
+Define server_name  buddysystem.local
```
Comment (JeremieSamson) [inline .docker/config/vhosts/buddysystem.conf]:
```suggestion
Define server_name buddysystem.local
```

File: src/Entity/UserHistory.php [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    {
+        return $this->nbBuddiesRefused;
+    }
+
+    public function setNbBuddiesRefused(int $nbBuddiesRefused): void
+    {
+        $this->nbBuddiesRefused = $nbBuddiesRefused;
+    }
+
+    public function getUser()
```
Comment (JeremieSamson) [inline src/Entity/UserHistory.php]: typehint

File: src/EventListener/ReportListener.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+class ReportListener
+{
+    public function prePersist(LifecycleEventArgs $args): void
+    {
+        $entity = $args->getObject();
+
+        if ($entity instanceof Report) {
+            $mentor = $entity->getMentor();
+            $mentor->setNbReport($mentor->getNbReport() + 1);
```
Comment (JeremieSamson) [inline src/EventListener/ReportListener.php]: increment ?

File: src/EventListener/ReportListener.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        }
+    }
+
+    public function preRemove(LifecycleEventArgs $args): void
+    {
+        $entity = $args->getObject();
+
+        if ($entity instanceof Report) {
+            $mentor = $entity->getMentor();
+            $mentor->setNbReport($mentor->getNbReport() - 1);
```
Comment (JeremieSamson) [inline src/EventListener/ReportListener.php]: decrement ?

File: src/Model/BuddyListResponse.php [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+            if (array_key_exists('section', $options)) {
+                $this->section = $options['section'];
+            }
+        }
+
+        return $this->getArrayResponse();
+    }
+
+    public function getCreatedAt()
```
Comment (JeremieSamson) [inline src/Model/BuddyListResponse.php]: remove getters plz

File: src/Model/UserListResponse.php [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+            if (array_key_exists('section', $options)) {
+                $this->section = $options['section'];
+            }
+        }
+
+        return $this->getArrayResponse();
+    }
+
+    public function getFirstname()
```
Comment (JeremieSamson) [inline src/Model/UserListResponse.php]: remove plz

File: src/Repository/BuddyRepository.php [hunk 8]
```diff
@@ -114,4 +141,78 @@ public function getBuddiesToWarn()
...
     }
+
+    public function getBuddiesByListUsers($idUsers)
+    {
+        $qb = $this->createQueryBuilder('b')
+            ->distinct()
+            ->innerJoin('b.mentor', 'mentor')
+            ->innerJoin('b.mentee', 'mentee')
+        ;
+        $qb->where($qb->expr()->in('mentor.id', $idUsers))
```
Comment (JeremieSamson) [inline src/Repository/BuddyRepository.php]:
```suggestion
        $qb
            ->where($qb->expr()->in('mentor.id', $idUsers))
```

File: src/Repository/UserRepository.php [hunk 9]
```diff
@@ -15,32 +17,37 @@ public function __construct(ManagerRegistry $registry)
...
+
+        return $qb->getQuery()->getSingleScalarResult();
+    }
+
     public function getInternationalStudentsNotMatched(Section $section)
     {
         $qb = $this->getStudentsNotMatched($section);
-        $qb->andWhere('u.isLocal = :isLocal')
-            ->setParameter('isLocal', false)
+        $qb->andWhere('u.local = :local')
```
Comment (JeremieSamson) [inline src/Repository/UserRepository.php]:
```suggestion
        $qb
            ->andWhere('u.local = :local')
```

File: src/Repository/UserRepository.php [hunk 10]
```diff
@@ -49,19 +56,79 @@ public function getInternationalStudentsNotMatched(Section $section)
     public function getLocalStudentsNotMatched(Section $section)
     {
         $qb = $this->getStudentsNotMatched($section);
-        $qb->andWhere($qb->expr()->eq('u.isLocal', ':isLocal'))
-            ->setParameter('isLocal', true)
+        $qb->andWhere($qb->expr()->eq('u.local', ':local'))
```
Comment (JeremieSamson) [inline src/Repository/UserRepository.php]:
```suggestion
        $qb
            ->andWhere($qb->expr()->eq('u.local', ':local'))
```

Comment (victor-paumier) [reply src/Repository/UserRepository.php]: done

File: src/Repository/UserRepository.php (hunk 9 above)
Comment (victor-paumier) [reply src/Repository/UserRepository.php]: done

File: src/Repository/BuddyRepository.php (hunk 8 above)
Comment (victor-paumier) [reply src/Repository/BuddyRepository.php]: done

File: src/Model/UserListResponse.php (hunk 7 above)
Comment (victor-paumier) [reply src/Model/UserListResponse.php]: done

File: src/Model/BuddyListResponse.php (hunk 6 above)
Comment (victor-paumier) [reply src/Model/BuddyListResponse.php]: done

File: src/EventListener/ReportListener.php (hunk 5 above)
Comment (victor-paumier) [reply src/EventListener/ReportListener.php]: done

File: src/EventListener/ReportListener.php (hunk 4 above)
Comment (victor-paumier) [reply src/EventListener/ReportListener.php]: done

File: src/Entity/UserHistory.php (hunk 3 above)
Comment (victor-paumier) [reply src/Entity/UserHistory.php]: done

File: .docker/config/vhosts/buddysystem.conf (hunk 2 above)
Comment (victor-paumier) [reply .docker/config/vhosts/buddysystem.conf]: done

File: .docker/config/vhosts/buddysystem.conf (hunk 1 above)
Comment (victor-paumier) [reply .docker/config/vhosts/buddysystem.conf]: done

**Pull request merged**